services/src/main/java/org/keycloak/services/managers/ClientManager.java

```diff
@@ -93,7 +94,8 @@ public static ClientModel createClient(KeycloakSession session, RealmModel realm
 
 
     public boolean removeClient(RealmModel realm, ClientModel client) {
-        if (realm.removeClient(client.getId())) {
+        if (!isInternalClient(realmManager.getSession().realms().getRealmsStream(), client.getClientId())
+                && realm.removeClient(client.getId())) {
```

Comment:
There should be check that the realm is actually master realm.

Reply:
Fixed. The "isInternalClient()" method now instantly returns false if it's the master realm.